operations/model.go

```diff
@@ -155,9 +159,50 @@ func (s *ClientSettings) setupRestCommunicator(ctx context.Context, printMessage
 	if printMessages {
 		printUserMessages(ctx, c, !s.AutoUpgradeCLI)
 	}
+
+	if s.shouldGenerateJWT(ctx, c) {
+		grip.Info("We will attempt to generate a JWT token, to opt out of this, set 'do_not_run_kanopy_oidc' to true in your config file")
+		if s.JWT, err = RunKanopyOIDCLogin(); err != nil {
+			grip.Warningf("Failed to get JWT token: %s", err)
+			return c, err
+		}
+		c.SetJWT(s.JWT)
+	}
+
 	return c, nil
 }
 
+func (s *ClientSettings) shouldGenerateJWT(ctx context.Context, c client.Communicator) bool {
+	if s.DoNotRunKanopyOIDC {
+		return false
+	}
+
+	if s.APIKey == "" {
+		grip.Info("No API key found, attempting to use a JWT token.")
```

Comment:
nit: might help to clarify to users who probably don't know much about their setup that the API key comes from their evergreen.yml file. Something like `No API key found in personal Evergreen YAML, ...`